### PR TITLE
Story #345 NET_DEBIT_CAP management, story #373 coverage tests 

### PR DESCRIPTION
**Pull request by shashi165**
**Story #345 Implement position API to manage NET DEBIT CAP**
1. API to get the existing limits
URL: GET http://{{BASE_URL}}/participants/{name}/limits?currency={currencyId}&type={typeValue}
Retuns:
```
[
    {
        "currency": "USD",
        "limit": {
            "type": "NET_DEBIT_CAP",
            "value": 30000000
        }
    },
    {
        "currency": "USD",
        "limit": {
            "type": "TEST",
            "value": 99999
        }
    }
]
```
2. API to Add/Update limit
URL: POST http://{{BASE_URL}}/participants/{name}/limits
Payload:
```
{
    "currency": "USD",
    "limit": {
    	"type": "NET_DEBIT_CAP",
        "value": 99999
    }
}

```
Return: 
`Status Code: 200`
Payload 
```
{
    "currency": "USD",
    "limit": {
        "type": "NET_DEBIT_CAP",
        "value": 99999
   }
}
```

**Story #373 unit/coverage tests for transactions**
 Added the unit tests and the coverage is 100% now

**Coverage Tests**

```
=============================== Coverage summary ===============================
Statements   : 65.24% ( 1642/2517 )
Branches     : 55.81% ( 293/525 )
Functions    : 51.17% ( 241/471 )
Lines        : 65.78% ( 1632/2481 )
================================================================================

1..619
# tests 619
# pass  619

# ok

```

**Unit Tests**
```
1..619
# tests 619
# pass  619

# ok

```

**Integration Tests**
```
1..107
# tests 107
# pass  107

# ok

```

### CODE REVIEW

File: src/admin/participants/handler.js [hunk 1]
```diff
@@ -138,12 +162,43 @@ const addLimitAndInitialPosition = async function (request, h) {
...
+    return limits
+  } catch (err) {
+    throw Boom.badRequest()
+  }
+}
+
+const adjustLimits = async function (request, h) {
+  Sidecar.logRequest(request)
+  try {
+    await Participant.adjustLimits(request.params.name, request.payload)
```
Comment (mdebarros) [inline src/admin/participants/handler.js]: Are we only catering for a single limit and not a list of limits?

I feel there is inconsistency here between the GET and the POST. Where the GET returns a list and the POST expects a single limit.

This does break the REST design paradigm.

Comment (shashi165) [reply src/admin/participants/handler.js]: Modified the GET to match with POST object
For now we are not considering the payload to be an array of multiple limits for POST, @NicoDuvenage  will create a separate story for that.

File: src/models/participant/facade.js [hunk 2]
```diff
@@ -221,10 +227,149 @@ const addLimitAndInitialPosition = async (participantCurrencyId, limitPostionObj
...
+        throw err
+      }
+    })
+  } catch (err) {
+    throw new Error(err.message)
+  }
+}
+
+/**
+ * @function GetParicipantLimitsByCurrencyId
```
Comment (rmothilal) [inline src/models/participant/facade.js]: Participant

File: src/models/participant/facade.js [hunk 3]
```diff
@@ -221,10 +227,149 @@ const addLimitAndInitialPosition = async (participantCurrencyId, limitPostionObj
...
+ * @description This retuns the active participant limits for a give participantCurrencyId and limit type
+ *
+ *
+ * @param {integer} participantCurrencyId - the id of the participant currency in the database. Example 1
+ * @param {string} type - The type of the limit. Example 'NET_DEBIT_CAP'
+ *
+ * @returns {array} - Returns an array containing the list of all active limits for the participant/currency and type if successful, or throws an error if failed
+ */
+
+const getParicipantLimitsByCurrencyId = async (participantCurrencyId, type) => {
```
Comment (rmothilal) [inline src/models/participant/facade.js]: Participant

File: src/models/participant/facade.js [hunk 4]
```diff
@@ -221,10 +227,149 @@ const addLimitAndInitialPosition = async (participantCurrencyId, limitPostionObj
...
+ * @description This retuns all the active endpoints for a give participantId  and limit type
+ *
+ *
+ * @param {integer} participantId - the id of the participant currency in the database. Example 1
+ * @param {string} type - The type of the limit. Example 'NET_DEBIT_CAP'
+ *
+ * @returns {array} - Returns an array containing the list of all active limits for the participant and type if successful, or throws an error if failed
+ */
+
+const getParicipantLimitsByParicipantId = async (participantId, type) => {
```
Comment (rmothilal) [inline src/models/participant/facade.js]: Participant

File: src/admin/participants/handler.js (hunk 1 above)
Comment (shashi165) [reply src/admin/participants/handler.js]: Changes/Updates
1. changed the output of GET
2. changed the output of POST
3.  updated the sequence diagram for POST
4. created the sequence diagram for GET
5. raised PR for these 2 diagrams "https://github.com/mojaloop/central-ledger/pull/103"
6. added few more coverage tests
7. created the story #383 to cater to multiple limits in POST

**Pull request merged**